app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -14,8 +14,15 @@ def sync_monitor
   end
 
   def sync_from_system
-    ensure_kube_monitors_started
+    # All miq_server instances have to reside on the same Kubernetes cluster, so
+    # we only have to sync the list of pods and deployments once
+    ensure_kube_monitors_started if my_server_is_primary?
```

Comment:
With this change I think we now need the "default" to be first in the list

Reply:
Actually, as I'm looking at all of this closer, this is all really weird to me.  We should really be starting this thread outside of the `as_each_server { monitor }` loop and also ensure it's still alive outside of the loop.  This whole monitoring kube thing has nothing to do with individual servers.  The only thing we really need individual servers for is to know if that server's workers are now orphaned or not.  Even then we don't need to do it server by server, because we can just make it check all the workers `in_my_region`.  I'm tempted to say let's just get rid of this whole "looping over the servers" thing.

Reply:
Oh, good point.  I don't think we need anything to be done on each individual server.  The code outside of the `as_each_server` block could easily check the deployments, settings, and pods/rows for the whole region.

Reply:
Yeah we're definitely overloading the miq_server concept here, doesn't help that we're creating multiple miq_servers for a single "platform" instance (aka before 1 miq_server <=> 1 vm)

I think we need to decide what exactly an miq_server is, does is track 1:1 with a runtime platform (vm, k8s namespace) or is it just a logical grouping for zones and other resources.

Reply:
Okay, short term I've unshifted the "real" miq_server to the front of the line.  Long term we definitely need to either extract this out of miq_server or not create "fake" miq_servers for zones.

Not sure if there is enough here to be considered short-term, maybe we just jump straight in to reworking this

Reply:
yeah I have a solid idea on my head on that - I'll rip out a quick WIP PR once this is merged to discuss.